### DISCUSSION TITLE
Problems in meshing

### DISCUSSION THREAD

**Discussion by acbof** [Ifsttar/I-Simpa#308] [Q&A]
Hello, 

I'm trying to run a simulation and keep getting two errors:

First, when it runs the tetgen: 
tetgen.exe : A self-intersection was detected. Program stopped.
tetgen.exe : Hint: use -d option to detect all self-intersections.

And then, when it runs the spps code (in random mode):
spps.exe :Tetrahedron file is empty, the calculation can't be done !
spps.exe :Unable to read the tetrahedalization of the scene mesh file, calculation canceled.

I know the first error could be due to 3D modeling problems, and I believe the second comes from the first. 

However, this happens in simulation that I have previously runned without any problems. Whenever I close I-Simpa and re-open it, this happens. It started happening after I tried to run two simulations at the same time. So, for every simulation file I have, I have to run every case without closing the software. 

Already tried reinstalling I-Simpa, but it didn't work. Do you have any idea of how I could solve this?

![image](https://user-images.githubusercontent.com/29489885/155707309-654fd3cd-6d98-4deb-8f63-c787d1c71fcb.png)

**Comment by nicolas-f** [comment]
Hi,

There is an issue with your 3D model. Duplicate face or self intersection issues. 

You can display the location of issues by following the guide:

https://i-simpa-wiki.readthedocs.io/fr/latest/import_file_recommandations.html#use-tetgen-to-identify-problems-in-3d-models

**Reply by acbof** [reply]
Ok, so I had two fitting zones intersecting with faces. I solved that, but now the software doesn't generate the mesh. It stays forever in the loading screen and does not make any progress. 
![image](https://user-images.githubusercontent.com/29489885/155714699-f11e5031-16a9-44a3-a4c8-1c25f4341b64.png)

Here's the console: 
![image](https://user-images.githubusercontent.com/29489885/155714940-26c20fdc-1093-4950-8275-d1a33c2692aa.png)

**Reply by nicolas-f** [reply]
You can uncheck the “Test mesh topology” option now

**Reply by acbof** [reply]
I did. The first screen print is what happens when I try to run the simulation with it unchecked.

**Reply by nicolas-f** [reply]
Its weird, it says there is no output file of tetgen

At first just try to visualize the meshing in I-Simpa
https://i-simpa-wiki.readthedocs.io/fr/latest/toolbars.html#toolbar-meshing
